client/src/components/common/download_files/DeleteModal.tsx

```diff
@@ -10,13 +9,29 @@ import { Modal, Button } from 'react-bootstrap';
  * @property {Function} onCancel  Called if delete cancelled
  */
 
+export interface DeleteModalProps {
+  /** Name of object to be deleted */
+  name: string;
+  /** Whether to show the model */
+  show: boolean;
+  /** Called if delete confirmed */
+  onDelete: () => void;
+  /** Called if delete cancelled */
+  onCancel: () => void;
+}
+
 /**
  * Delete confirmation modal component
  *
- * @param {DeleteModalProps} props Props
- * @returns {React.Component<DeleteModalProps>} Component
+ * @param props Props
+ * @returns Component
  */
-export default function DeleteModal({ name, show, onDelete, onCancel }) {
+export default function DeleteModal({
+  name,
+  show,
+  onDelete,
+  onCancel,
+}: DeleteModalProps) {
```

Comment:
I've recently seen a suggestion that when we can use type inference for local variables, etc. but to explicitly type function declarations. I think it would be a good practise to declare `: JSX.Element` here just to avoid any weird type inference propagation in the future.

Reply:
Seems like a reasonable idea!